update mcp-gateway authn/authz guides (#110677)
openshift/openshift-docs: modules/proc-mcp-gateway-authorization.adoc

* You installed {mcpg}.
* You installed {prodname}.
* You configured a `Gateway` object with an `mcp` listener.
* You completed authentication procedures, including creating an `AuthPolicy` CR on the `mcp` listener.
* You configured your identity provider to include `group` and `role` claims in JSON Web Tokens (JWT).
* The identity provider client IDs match the namespaced `MCPServerRegistration` name in the format `_<namespace>_/_<mcpserverregistration_name>_`.

.Procedure

. Add an `mcps` listener to the `Gateway` object for internal `tools/call` routing by using the following command as an example:
+
[source,json,subs="+quotes"]
----

> $ oc patch gateway _<mcp_gateway>_ -n _<gateway_system>_ --type json -p '[

Reviewer: should this section, lines [44-50](https://github.com/openshift/openshift-docs/pull/109364/changes#diff-cd246969df3ee03d8d2c949fefa3874ac35a02058eb03c33758d3d6c82409517R44) in the install doc be updated to match this? 

- we can make the listener a prereq to this procedure
- add a note to the install doc that you need to config this listener to create authorization
- then we can remove the step from this procedure

WDYT?

Author: sounds good to me, I removed it from this pr and added it as a prereq. Do you want to add it to your current installation PR, or will I create a follow-up PR?

Reviewer: I can add it to the installation PR, but since you removed it from this PR I can't see it, so I might miss something.

}
----
+

> * The `"mcp-ns/arithmetic-mcp-server"` specification must match the namespaced name of the `MCPServerRegistration` CR in the format `{namespace}/{name}`. For example, if your `MCPServerRegistration` is named `arithmetic-mcp-server` in the `mcp-ns` namespace, the {keycloak} client ID must be `mcp-ns/arithmetic-mcp-server`.

Reviewer: ```suggestion
* The `"mcp-ns/arithmetic-mcp-server"` specification must match the namespaced name of the `MCPServerRegistration` CR in the format `{namespace}/{name}`. For example, if your `MCPServerRegistration` CR is named `arithmetic-mcp-server` and is applied in the `mcp-ns` namespace, the {keycloak} client ID must be `mcp-ns/arithmetic-mcp-server`.
```

* The `"roles": ["add", "sum", "multiply", "divide"]` parameter and values specify the roles representing the allowed tools.

. Configure tool-level authorization by creating an `AuthPolicy` CR that enforces tool-level access control on the `mcps` listener, as shown in the following example:
+
[IMPORTANT]
====

> The authorization `AuthPolicy` must target the `mcps` listener, not the `mcp` listener. The `mcp` listener handles public traffic (`initialize`, `tools/list`, `/.well-known`) and has the authentication-only `AuthPolicy`.

Reviewer: ```suggestion
The authorization `AuthPolicy` CR must target the `mcps` listener, not the `mcp` listener. The `mcp` listener only handles public traffic and has an authentication-only `AuthPolicy` CR.
```

* Replace `metadata.name:` with the name of the `AuthPolicy`.
* Replace `metadata.namespace:` with the namespace where the `AuthPolicy` CR is applied.
* Replace `spec.targetRef.name:` with the name of the `Gateway` CR.

> * The `spec.targetRef.sectionName:` value must be `mcps`, the internal listener for tool-call authorization. This listener must exist on your `Gateway` object.

Reviewer: ```suggestion
* The `spec.targetRef.sectionName:` value must be `mcps`, which is the internal listener for `tool/call` authorization. This listener must exist on your `Gateway` object.
```

+
[source,terminal,subs="+quotes"]
----
$ oc apply -f - <<EOF

> <AuthPolicy CR from the previous step>

Reviewer: ```suggestion
_<mcp_tool_auth_policy.yaml>_
```
user-replaceable values can only look like this
I also don't think the EOF works when we separate file creation from application? (having the yaml in one step and application of the resource in a second step is preferred in docs)
